data/wbd/clip_vectors_to_wbd.py

```diff
@@ -102,6 +102,20 @@ def extend_outlet_streams(streams, wbd_buffered, wbd):
         else:
             errors += 1
 
+        # Snap levelpath endpoints (mainstem only) to the nearest landsea boundary
+        # if they are within 10 km, ensuring outlets align with the landsea.
+        if landsea is not None:
+            landsea_union = landsea.geometry.unary_union
+            levelpath_outlets = levelpath_outlets[levelpath_outlets['mainstem'] == 1]
+            for index, row in levelpath_outlets.iterrows():
+                endpoint = Point(row['geometry'].coords[-1])
+                nearest_point_landsea = nearest_points(endpoint, landsea_union)[1]
+                distance = endpoint.distance(nearest_point_landsea)
+                if distance <= 10000:
+                    coords = list(row['geometry'].coords)
+                    coords[-1] = nearest_point_landsea.coords[0]
```

Comment:
It would probably be better to do something like `coords += nearest_point_landsea.coords[0]` to avoid overwriting and discarding the previous last point.